### PR TITLE
Add streaming support

### PR DESCRIPTION
**Pull request by daulet**
- [x] Decide on exposing multiple generations: multiple channels or one for all?
- [x] Tests
- [x] Add example usage to README

## Usage:
```go
	prompt := `Write 5 titles for a blog ideas for the keywords "large language model" or "text generation"`
	ch := co.Stream(cohere.GenerateOptions{
		Model:       "xlarge",
		Prompt:      prompt,
		MaxTokens:   cohere.Uint(200),
		Temperature: cohere.Float64(0.75),
	})

	fmt.Printf("Completion:\n%s", prompt)
	for res := range ch {
		if res.Err != nil {
			fmt.Println(res.Err)
			break
		}
		fmt.Printf(res.Token.Token)
	}
```

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -97,14 +97,43 @@ func (c *Client) post(endpoint string, body interface{}) ([]byte, error) {
 	return buf, nil
 }
 
+func (c *Client) stream(endpoint string, body interface{}) (io.ReadCloser, error) {
```
Comment (miguelvr) [inline client.go:100]: isn't body always of type `GenerationOptions`?

Comment (daulet) [reply client.go:100]: no, that's the other `Stream()`. I'm thinking if I should just make this strongly typed too.

Comment (daulet) [reply client.go:100]: nvm, you are right, i thought we had dedicated request type, we don't

**Comment by daulet** [comment]
@miguelvr what do you think of API for multiple generations, i.e. when `num_generations > 1`? Do we want to expose those as distinct channels, i.e. `Stream()` returns a slice of channels, or do we keep it simple and send everything over the single channel?

**Review by 1vn — APPROVED** [review]
LGTM

**Comment by daulet** [comment]
suggesting to merge this without final message that will have more info, but that's currently blocked.

**Review by lfayoux — APPROVED** [review]
lgtm

**Pull request merged**